src/validation.cpp

```diff
@@ -609,17 +609,66 @@ static bool AcceptToMemoryPoolWorker(const CChainParams& chainparams, CTxMemPool
+            // very realistic either, thus we we try to ensure a limited set of transactions are RBF'able despite
+            // mempool conflicts here. Importantly, we want to ensure that transactions which were accepted using
+            // the below carve-out are able to be RBF'ed, without impacting the security the carve-out provides
+            // for off-chain contract systems (see link in the comment below).
+            //
+            // Specifically, the subset of RBF transactions which we allow despite chain limits are those which
+            // conflict directly with exactly one other transaction (but may evict children of said transaction),
+            // and which are not adding any new mempool dependencies (checked first).
+            // Such transactions are clearly not merging any existing packages, so we are only concerned with
+            // ensuring that (a) no package is growing past the package size (not count) limits and (b) we are
+            // not allowing something to effectively use the (below) carve-out spot when it shouldn't be allowed
+            // to.
+            //
+            // To check these we first check if we meet the RBF criteria, above, and increment the descendant
+            // limits by the single direct conflict (as these are recalculated by assuming the new transaction
+            // being added is a new dependant, with no removals, of each parents' existing dependant set).
+            // The ancestor count limits are unmodified (as the ancestor limits should be the same for both our
+            // new transaction and any conflicts).
+            assert(setIterConflicting.size() == 1);
+            CTxMemPool::txiter conflict = *setIterConflicting.begin();
+
+            std::set<uint256> conflict_existing_mempool_inputs;
+            for (const CTxMemPool::txiter& txit : pool.GetMemPoolParents(conflict)) {
+                conflict_existing_mempool_inputs.insert(txit->GetTx().GetHash());
+            }
+
+            bool meets_rbf_conflict_limit = true;
+            for (const CTxIn& new_input: tx.vin) {
+                if (conflict_existing_mempool_inputs.count(new_input.prevout.hash) == 0) {
+                    if (pool.exists(new_input.prevout.hash)) {
+                        meets_rbf_conflict_limit = false;
+                    }
+                }
+            }
+
+            if (meets_rbf_conflict_limit) {
+                nLimitDescendants += 1;
+                nLimitDescendantSize += conflict->GetTxSize();
+            }
+        }
+
         std::string errString;
         if (!pool.CalculateMemPoolAncestors(entry, setAncestors, nLimitAncestors, nLimitAncestorSize, nLimitDescendants, nLimitDescendantSize, errString)) {
             setAncestors.clear();
             // If CalculateMemPoolAncestors fails second time, we want the original error string.
             std::string dummy_err_string;
+            // Contracing/payment channels CPFP carve-out:
```

Comment:
"Contracting" with a t? :)

Reply:
No one ever warned me I'd need to know how to spell to write code :(